File: test_py/test_utils.py
import pytest
from src_py.grndwork_api_client.utils import strip_uuid


def describe_strip_uuid():
    @pytest.mark.parametrize('filename,expected', [
        ('', ''),
        ('Report', 'Report'),
        ('Report.pdf', 'Report.pdf'),
        ('Report_1234.pdf', 'Report_1234.pdf'),
        ('Report_1234abcd-1234-abcd-1234-abcd1234abcd', 'Report'),
        ('Report_1234abcd-1234-abcd-1234-abcd1234abcd.pdf', 'Report.pdf'),
        ('Report1234abcd-1234-abcd-1234-abcd1234abcd.pdf', 'Report.pdf'),
    ])
    def it_returns_filename_without_uuid(filename, expected):
        assert strip_uuid(filename) == expected

File: test_js/test_utils.ts
import {stripUUID} from '../src_js/grndwork_api_client/utils';

describe('stripUUID', () => {
  it.each([
    ['', ''],
    ['Report', 'Report'],
    ['Report.pdf', 'Report.pdf'],
    ['Report_1234.pdf', 'Report_1234.pdf'],
    ['Report_1234abcd-1234-abcd-1234-abcd1234abcd', 'Report'],
    ['Report_1234abcd-1234-abcd-1234-abcd1234abcd.pdf', 'Report.pdf'],
    ['Report1234abcd-1234-abcd-1234-abcd1234abcd.pdf', 'Report.pdf'],
  ])('returns filename without uuid', (filename, expected) => {
    expect(stripUUID(filename)).toEqual(expected);
  });
});
